### PR TITLE
feat: configurable products altogether

### PR DESCRIPTION
**Pull request by ivan-kalachikov**
## Description
image
image
image
image

### Jira-link:
https://virtocommerce.atlassian.net/browse/VCST-2255
### Artifact URL:
https://vc3prerelease.blob.core.windows.net/packages/vc-theme-b2b-vue-2.10.0-pr-1440-322a-322aa4aa.zip

### CODE REVIEW

File: client-app/shared/cart/components/add-to-cart.vue [hunk 1]
```diff
@@ -100,21 +106,23 @@ async function onChange() {
...
+  const mode = isAlreadyExistsInTheCart && !isConfigurable.value ? AddToCartModeType.Update : AddToCartModeType.Add;
+
+  if (mode === AddToCartModeType.Update) {
     updatedCart = await changeItemQuantity(lineItem!.id, enteredQuantity.value || 0);
   } else {
     const inputQuantity = enteredQuantity.value || minQty.value;
-
-    updatedCart = await addToCart(props.product.id!, inputQuantity);
+    const configurationSections = isConfigurable.value ? selectedConfigurationInput.value : undefined;
+    updatedCart = await addToCart(product.value.id!, inputQuantity, configurationSections);
```
Comment (NaMax66) [inline client-app/shared/cart/components/add-to-cart.vue:116]: remove !

File: client-app/shared/catalog/components/product-configuration.vue [hunk 2]
```diff
@@ -6,55 +6,104 @@
...
     <div class="product-configuration__widgets">
-      <VcWidget collapsible size="xs">
+      <VcWidget v-for="section in configuration" :key="section.id" collapsible size="xs">
         <template #title>
-          Title
+          {{ section.name }}
           <div class="product-configuration__description">
-            Colorful Party Balloons 100pcs 12inch Chrome Metallic Helium Balloons for Birthday Party Decoration and Arch
-            Decoration
+            {{ getSelectedOptionTitle(section.id!) ?? section.description }}
```
Comment (NaMax66) [inline client-app/shared/catalog/components/product-configuration.vue:13]: remove ! or change getSelectedOptionTitle

File: client-app/shared/catalog/components/product-configuration.vue [hunk 3]
```diff
@@ -6,55 +6,104 @@
...
           </div>
         </template>
 
-        <div class="product-configuration__items">
-          <VcProductCard v-for="i in ['1', '2', '3']" :key="i" view-mode="item" class="product-configuration__item">
+        <div v-if="section.type === 'product' && section.options?.length" class="product-configuration__items">
+          <template
+            v-for="({ id, product, quantity, listPrice, salePrice, extendedPrice }, index) in section.options"
+            :key="`${id}-${index}`"
```
Comment (NaMax66) [inline client-app/shared/catalog/components/product-configuration.vue:20]: why ${id}-${index}`? 
keep id or index

File: client-app/shared/catalog/components/product-configuration.vue [hunk 4]
```diff
@@ -6,55 +6,104 @@
...
+              <VcProductTitle :to="getProductRoute(product.id, product.slug)">{{ product.name }}</VcProductTitle>
+
+              <VcProductProperties>
+                <VcProperty
+                  v-for="property in product.properties.slice(0, PRODUCT_PROPERTY_LIMIT)"
+                  :key="property.id"
+                  :label="property.name"
+                  :value="property.value"
+                />
+                <VcProperty class="@2xl:hidden" label="Price per item">
```
Comment (NaMax66) [inline client-app/shared/catalog/components/product-configuration.vue:45]: :label="$t(..)"

File: client-app/shared/catalog/components/product-configuration.vue [hunk 5]
```diff
@@ -6,55 +6,104 @@
...
+            </VcProductCard>
+          </template>
+
+          <VcProductCard v-if="!section.isRequired" view-mode="item" class="product-configuration__item">
             <template #media>
-              <VcRadioButton :value="i" name="radio" />
+              <VcRadioButton
+                value="none"
+                :name="`selection-${section.id}`"
+                :model-value="selectedConfiguration[section.id!]?.productId === undefined ? 'none' : ''"
```
Comment (NaMax66) [inline client-app/shared/catalog/components/product-configuration.vue:66]: section.id! -> section.id

File: client-app/shared/catalog/composables/useConfigurableProduct.ts [hunk 6]
```diff
@@ -120,15 +142,19 @@ export function useConfigurableProduct(configurableProductId: string) {
...
+      immediate: true,
     },
   );
 
   return {
     fetchProductConfiguration,
     selectSectionValue,
     loading: computed(() => fetching.value || creating.value),
-    configuration: readonly(configuration),
+    configuration: shallowReadonly(configuration),
```
Comment (NaMax66) [inline client-app/shared/catalog/composables/useConfigurableProduct.ts:153]: lets keep just readonly. Less questions. Or if needed - add setter to the composable

File: client-app/shared/catalog/composables/useConfigurableProduct.ts [hunk 7]
```diff
@@ -55,19 +61,24 @@ export function useConfigurableProduct(configurableProductId: string) {
...
+          changeSelectionValue({
+            sectionId: section.id,
+            value: {
+              productId: section.options?.[0]?.product?.id ?? "",
+              quantity: section.options?.[0]?.quantity ?? 1,
+            },
+          });
+        }
+      });
+      await nextTick();
```
Comment (NaMax66) [inline client-app/shared/catalog/composables/useConfigurableProduct.ts:80]: dirty hack?

**Review by NaMax66 — APPROVED** [review]

File: client-app/shared/catalog/composables/useConfigurableProduct.ts (hunk 7 above)
Comment (ivan-kalachikov) [reply client-app/shared/catalog/composables/useConfigurableProduct.ts:80]: just forgot to remove that after some experiments :) thanks for pointing this out

**Pull request merged**